etna/models/nn/chronos/base.py

```diff
@@ -0,0 +1,277 @@
+        device: str,
+        dtype: torch.dtype,
+        cache_dir: Path,
+    ):
+        """
+        Init Chronos-like model.
+
+        Parameters
+        ----------
+        path_or_url:
+            Path to the model. It can be huggingface repository, local path or external url
+            - If huggingface repository, see available models in ``list_models`` of appropriate model class.
+            During the first initialization model is downloaded from huggingface and saved to local ``cache_dir``.
+            All following initializations model will be loaded from ``cache_dir``. See ``pretrained_model_name_or_path`` parameter of :py:func:`transformers.PreTrainedModel.from_pretrained`
+            - If local path, model will not be saved to local ``cache_dir``.
+            - If external url, it must be zip archive with the same name as model directory inside. Model will be downloaded to ``cache_dir``.
+        encoder_length:
+            Number of last timestamps to use as a context.
+        device:
+            Device type. See ``device_map`` parameter of :py:func:`transformers.PreTrainedModel.from_pretrained`.
+        dtype:
+            Torch dtype of computation. See ``torch_dtype`` parameter of :py:func:`transformers.PreTrainedModel.from_pretrained`.
+        cache_dir:
+            Local path to save model from huggingface during first model initialization. All following class initializations appropriate model version will be downloaded from this path.
+            See ``cache_dir`` parameter of :py:func:`transformers.PreTrainedModel.from_pretrained`.
+        """
+        super().__init__()
+        self.path_or_url = path_or_url
+        self.encoder_length = encoder_length
+        self.device = device
+        self.dtype = dtype
+        self.cache_dir = cache_dir
+
+        if self._is_url():
+            full_model_path = self._download_model_from_url()
+            self.pipeline = BaseChronosPipeline.from_pretrained(
+                full_model_path, device_map=self.device, torch_dtype=self.dtype, cache_dir=self.cache_dir
+            )
+        else:
+            self.pipeline = BaseChronosPipeline.from_pretrained(
+                self.path_or_url, device_map=self.device, torch_dtype=self.dtype, cache_dir=self.cache_dir
+            )
+
+    def _is_url(self):
+        return self.path_or_url.startswith("https://") or self.path_or_url.startswith("http://")
+
+    def _download_model_from_url(self) -> str:
+        """Download model from url to local cache_dir."""
+        model_file = self.path_or_url.split("/")[-1]
+        model_dir = model_file.split(".zip")[0]
+        full_model_path = f"{self.cache_dir}/{model_dir}"
+        if not os.path.exists(full_model_path):
+            request.urlretrieve(url=self.path_or_url, filename=model_file)
+
+            with zipfile.ZipFile(model_file, "r") as zip_ref:
+                zip_ref.extractall(self.cache_dir)
+            os.remove(model_file)
```

Comment:
This line won't work if we have some error during download or unpack. There are possible solutions
- Use temporary directory for loading file
- Wrap loading logic with `try` / `finally`